plugins/scaffolder/src/components/fields/RepoUrlPicker/AzureRepoPicker/plugin.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  ConfigApi,
+  createApiRef,
+  DiscoveryApi,
+} from '@backstage/core-plugin-api';
+
+import { Organization, Project } from './types';
+
+export interface ProxyAzureDevOpsPluginApi {
+  allowedOrganizations(): Promise<Organization[]>;
+  allowedProjects(organization: string): Promise<Project[]>;
+}
+
+export class ProxyAzureDevOpsPluginApiClient
+  implements ProxyAzureDevOpsPluginApi
```

Comment:
Hm neither the interface nor the class are exported publicly right? So the implementation isn't replaceable even though this goes through the effort of having an api ref etc?